build-debug.sh equivalent for cmake

ThaFireDragonOfDeath opened the pull request:
**Change description**
At the current time the cmake build process is a bit complicated and its not so easy to build and run Synfig Studio without installation.

This PR adds the possibility to make a portable cmake build with one command.
This can be helpful to test the cmake build process after bigger code changes.

**Building with this script**
1. cd to root dir of the synfig project (where the README is located)
2. ./build-cmake.sh (cleans the build dir and builds ETL, synfig-core and synfig-studio)
3. ./run-cmake.sh (runs synfig-studio)

**Other changes**
Fixed cmake build errors

**Tests**
Successfully build and run Synfig Studio on Manjaro Linux.

**Future**
The script can be improved on the following points:
- Build in Debug or Release mode
- Build single components
- Parallel building

I hope this helps and sorry for my bad English.

morevnaproject commented:
Hello! This is a very useful addition, thank you! I can see this PR also updates Cmake scripts to build with SDL2/SDL2_mixer. I need some time to review everything with consideration how this fit with #694.

A quick first notes:

1. I suggest to set build/install path to "_production" instead of "_cmake-build". The direction of our development is to migrate from Autotools to Cmake, so it will be good if Cmake already act as replacement.

2. Instead of "./run-cmake.sh" script I proposing to modify " autobuild/build-cmake-common.sh" so it will create "_production/build/synfigstudio.sh" and "_production/build/synfig.sh" scripts when everything is done.

ThaFireDragonOfDeath commented:
I just implemented the two points from the notes and some more.

**Changes with the last commits**
- Added parameters to control if its a debug or release build, parallel make jobs count, data prefix (where Synfig Studio search the pixmaps, sounds, plugins etc)
- Added: A run-portable.sh file will be written to the out directory after a successful build (can be disabled with the -n option of build-cmake). the script has the same function and usage as the old run-cmake.sh
- Changed the build directory to _debug / _production (depends if it is a debug or release build)
- Improved the documentation for this changes
- Removed the run-cmake.sh

morevnaproject commented:
@ThaFireDragonOfDeath Awesome, thank you! So, let's keep the ball rolling and merge this PR. Any further changes can be added as separate PRs. Cheers!
